nixos/modules/services/networking/wireguard.nix

```diff
@@ -309,6 +366,11 @@ let
         path = with pkgs; [ iproute2 wireguard-tools ];
 
         serviceConfig =
+          {
+            # Overcomes failure to resolve the endpoint's host
+            # (eg, when the resolver is not yet reachable).
+            Restart = "on-failure";
```

Comment:
Since the network can be down arbitrarily long, we should probably set

```
unitConfig.StartLimitIntervalSec = 0;
```

otherwise systemd will restart the service only a few times until the resetart limit is reached (see #53852, https://unix.stackexchange.com/questions/289629/systemd-restart-always-is-not-honored#comment691882_324297).

Reply:
Indeed, thanks!